### PR TITLE
Move error response deserialization into the runtime

### PR DESCRIPTION
**Pull request by daschult**
Resolves https://github.com/Azure/ms-rest-js/issues/97 (partially).

I'll move success response deserialization in a different PR.

### CODE REVIEW

File: src/vanilla/ClientModelExtensions.cs [hunk 1]
```diff
@@ -661,6 +567,26 @@ public static void ConstructRequestBodyMapper(TSValue value, ParameterTS request
...
+            {
+                value.Text($"Mappers.{responseBodyModelType.Name}");
+            }
+            else
+            {
+                ConstructMapper(
+                    value,
+                    responseBodyModelType,
+                    "parsedResponse",
+                    null,
```
Comment (RikkiGibson) [inline src/vanilla/ClientModelExtensions.cs]: Please label this and any of the above arguments that aren't self-explanatory based on the values/names

File: src/vanilla/Model/MethodTS.cs [hunk 2]
```diff
@@ -915,6 +748,34 @@ public void GenerateOperationSpec(TSObject operationSpec)
                 }
             }
 
+            operationSpec.ObjectProperty("responses", responses =>
+            {
+                bool isXml = CodeModelTS.ShouldGenerateXmlSerialization;
+                foreach (KeyValuePair<HttpStatusCode, Response> statusCodeResponsePair in Responses)
```
Comment (RikkiGibson) [inline src/vanilla/Model/MethodTS.cs]: `foreach((HttpStatusCode status, Response response) in Responses)`

File: src/vanilla/Model/MethodTS.cs [hunk 3]
```diff
@@ -750,70 +612,41 @@ public string GenerateWithHttpOperationResponseMethod(string emptyLine)
...
                         {
                             tryBlock.LineComment("Deserialize Response");
-                            tryBlock.If($"statusCode === {GetStatusCodeReference(responsePair.Key)}", ifBlock =>
+                            tryBlock.Line("let statusCode = operationRes.status;");
+                            tryBlock.Line(InitializeResult);
+                        }
+
+                        if (hasResponsePairs)
+                        {
+                            foreach (KeyValuePair<HttpStatusCode, Response> responsePair in responsePairs)
```
[the remark this answers is not in the document]
Comment (daschult) [reply src/vanilla/Model/MethodTS.cs]: Fixed in https://github.com/Azure/autorest.typescript/pull/177/commits/f49ffa6c69fab6da9cdaa26f1b3f60b4bc4f794c.

File: src/vanilla/ClientModelExtensions.cs (hunk 1 above)
Comment (daschult) [reply src/vanilla/ClientModelExtensions.cs]: Fixed in https://github.com/Azure/autorest.typescript/pull/177/commits/f49ffa6c69fab6da9cdaa26f1b3f60b4bc4f794c.

File: src/vanilla/Model/MethodTS.cs (hunk 2 above)
Comment (daschult) [reply src/vanilla/Model/MethodTS.cs]: Fixed in https://github.com/Azure/autorest.typescript/pull/177/commits/f49ffa6c69fab6da9cdaa26f1b3f60b4bc4f794c.

**Comment by RikkiGibson** [comment]
# 🤖 AutoRest automatic bundle size check 🤖
  ### Size decreased by 18.11% (-343.59 KB)

**Comment by RikkiGibson** [comment]
Brings us from 217 -> 211 kB in the NetworkInterfaces context case and 1.23 -> 1.12 mB for the monolithic arm-network case.

**Comment by daschult** [comment]
Too bad. I was hoping for a bigger drop than that.

**Pull request merged**